src/Control/Monad/Linear.hs

```diff
@@ -34,21 +37,27 @@ import Prelude (String)
 -- TODO: make the laws explicit
 
 -- | Enriched linear functors.
-class Functor f where
+class Data.Functor f => Functor f where
   fmap :: (a ->. b) ->. f a ->. f b
 
 (<$>) :: Functor f => (a ->. b) ->. f a ->. f b
 (<$>) = fmap
 
+dataFmapDefault :: Functor f => (a ->. b) -> f a ->. f b
+dataFmapDefault f = fmap f
+
 -- | Enriched linear applicative functors
-class Functor f => Applicative f where
+class (Data.Applicative f, Functor f) => Applicative f where
   {-# MINIMAL pure, ((<*>) | liftA2) #-}
   pure :: a ->. f a
   (<*>) :: f (a ->. b) ->. f a ->. f b
   (<*>) = liftA2 id
   liftA2 :: (a ->. b ->. c) ->. f a ->. f b ->. f c
   liftA2 f x y = f <$> x <*> y
 
+dataPureDefault :: Applicative f => a -> f a
+dataPureDefault x = pure x
```

Comment:
Maybe we could also straight up give `DerivingVia` combinator for these.

Like, define a data type

```haskell
newtype Forget f a = Forget (f a)
```

Then prove

```haskell
instance Functor f => Data.Forget (Control f)
instance Applicative f => Data.Applicative (Control f)
```

Then we can use it in `deriving` clauses.

Reply:
Do you mean 
```haskell
instance Functor f => Data.Functor (Forget f)
instance Applicative f => Data.Applicative (Forget f)
```

Reply:
Yes. Let's blame it on the heat.

Reply:
I've added this, with a different name - since the name is exported I felt something more evocative would be nicer.

Reply:
I wonder if it is worth adding 
```
Linear.coerce :: Coercible a b => a ->. b
```
also.